### ISSUE TITLE
NoSuchMethodError: Class 'String' has no instance getter 'statusCode'.

### ISSUE THREAD

**Issue by debasiskoley** [flutterchina/dio#331]
### New Issue Checklist

- [ ] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |      |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

500
E/flutter (20017): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: NoSuchMethodError: Class 'String' has no instance getter 'statusCode'.
E/flutter (20017): Receiver: "ErrorRes: {\"status\":false,\"data\":{\"message\":\" please try again later.\"},\"message\":\"error\"}"
E/flutter (20017): Tried calling: statusCode
E/flutter (20017): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)

**Comment by Fejick** [comment]
Can you plz send the code on how you were calling that function ?
